content/6_Advanced/Wavelet.mdx

```diff
@@ -32,9 +30,250 @@ Like a segment tree on values rather than indices.
 	</Resource>
 </Resources>
 
-### Solution - Range K-th Smallest
+Suppose you want to support the following queries:
+
+- Given a range $l$, $r$ count number of occurrences of value $x$.
+- Given a range $l$, $r$ find the $k$ smallest element
+
+With wavelet tree you can easily support those queries in $log(n)$ time.
+
+<Info title="Additional information">
+  It is possible to support with updates in $\mathcal{O}(\log n^2)$
+</Info>
+
```

Comment:
unless you give an impl or a link to an impl don't really think this is necessary

Reply:
I explained updates below.

Reply:
think it would be better to just remove this and talk about it below